### PR TITLE
User controller refactor

### PR DESCRIPTION
**Pull request by evansiroky**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

The user controller has a lot of repetitive code and does not have thorough logging. This PR consolidates a lot of common code and adds many more logging statements to help with debugging. This should work with a config with valid auth0 settings.

### CODE REVIEW

File: src/main/java/com/conveyal/datatools/manager/controllers/api/UserController.java [hunk 1]
```diff
@@ -327,6 +282,140 @@ private static Object getRecentActivity(Request req, Response res) {
...
+    /**
+     * Create a HTTP POST request with some default headers
+     */
+    private static HttpPost makePostRequest(String url) {
+        HttpPost request = new HttpPost(url);
+        request.addHeader("Authorization", "Bearer " + AUTH0_API_TOKEN);
+        request.setHeader("Accept-Charset", charset);
+        request.setHeader("Content-Type", "application/json");
+        return request;
+    }
```
Comment (landonreed) [inline src/main/java/com/conveyal/datatools/manager/controllers/api/UserController.java]: It seems like there is a lot of duplicated code in `makePostRequest` and `makeGetRequest`. Also, the same headers are set in `deleteUser`. Perhaps, this should all be refactored into a single function that takes as input an `HttpEntityEnclosingRequestBase` and sets headers the Auth0-related headers called, for example, `setAuth0Headers`. It probably wouldn't hurt to set content type for all request types just to simplify things (or you could only set that header for posts). Thoughts?

**Comment by evansiroky** [comment]
Just consolidated code as suggested.

**Comment by landonreed** [comment]
Thanks for making that change. I guess I'm unclear as to why `makeGetRequest` and `makePostRequest` would still need to exist now that we have `setHeaders`. At this point they just seem to be adding bloat to the code. I suppose technically `makePostRequest` is used twice, but it seems better to just have each controller method handle creating the HTTP request the same way.

**Comment by evansiroky** [comment]
Turns out it saves lines of code to not have those helper functions now.

**Pull request merged**